concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -2656,6 +2664,49 @@ handleUpdateCredentialKeys wtc cid keys sigs =
         updateCredentialKeys (fst senderAccount) index keys
         return (TxSuccess [CredentialKeysUpdated cid], energyCost, usedEnergy)
 
+-- | Handler for a token holder transaction.
+handleTokenHolder ::
+    ( PVSupportsPLT (MPV m),
+      SchedulerMonad m
+    ) =>
+    WithDepositContext m ->
+    -- | Token symbol identifying the token to receive the operations.
+    TokenId ->
+    -- | Operations for the token.
+    TokenParameter ->
+    m (Maybe TransactionSummary)
+handleTokenHolder depositContext tokenId tokenOperations =
+    withDeposit depositContext computeTransaction commitTransaction
+  where
+    senderAccount = depositContext ^. wtcSenderAccount
+    -- Execute the transaction in a modified environment
+    computeTransaction = do
+        -- Charge the base cost for this transaction type
+        tickEnergy Cost.tokenHolderBaseCost
+        -- Fetch the token from state
+        tokenIndex <-
+            lift (getTokenIndex tokenId) >>= \case
+                Just tokenIndex -> return tokenIndex
+                Nothing -> rejectTransaction $ NonExistentTokenId tokenId
+        -- Fetch the token configuration to find the reference for the token module.
+        configuration <- lift $ getTokenConfiguration tokenIndex
+        let moduleRef = Token._pltModule configuration
+        -- TODO Tick energy for loading the module into memory based on the module size. (Issue https://linear.app/concordium/issue/COR-1337)
+        -- Invoke the token module with operations.
+        events <- invokeTokenHolderOperations moduleRef tokenIndex senderAccount tokenOperations
```

Comment:
It would be good if this also handled the failure case.

Reply:
`invokeTokenHolderOperations` can reject the transaction directly.
Also not sure what type of reject reasons we intend for the token modules.